### DISCUSSION TITLE
I am using GitHub code search to find Angular versions based on a regex pattern. The goal is to categorize the results according to the identified Angular versions.

### DISCUSSION THREAD

**Discussion by kaushalgupta94** [community/community#102615] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

"I have numerous Angular repositories, each with different versions ranging from 4 to 16. I aim to refine my code search to retrieve repositories specifically for each version like for example -> version 5 with this query '/"@angular\/core": "[^]?5..*"/' works on the GitHub search page, 
it doesn't produce the expected results  (always count=0) when used with the GitHub REST API endpoint 'https://api.github.com/search/code?q=/"@angular\/core": "[^]?5..*"/ user:kgupta94'. I would appreciate guidance on correcting this issue."

**Comment by naveen-98** [comment]
To categorize Angular versions in GitHub repos:

Escape slashes: Use `\/` instead of `/ `in your regex pattern (e.g., `\/"@angular/core": "\[^]?5..\"').

Mind rate limits: Avoid hitting your API request limit, which can return empty results.

Double-check parameters: Ensure you're using the correct endpoint (`https://api.github.com/search/code`) and parameters (`q`, `user`).

Consider alternatives: Try GraphQL API, GitHub CLI, or custom parsing for fine-grained control.

Test your refined query directly on GitHub search first for accuracy. 

Good luck!

**Comment by kaushalgupta94** [comment]
Thanks @naveen-98 , 
I used this only what you suggested --> Escape slashes: Use \/ instead of / in your regex pattern (e.g., `/"@angular/core": "[^]?5.."'). 
actually , in the post body , its not showing , please refer to attached screenshot for actual post
<img width="698" alt="Screenshot 2024-01-29 205154" src="https://github.com/community/community/assets/41044729/eddfee8a-9696-4902-affc-c0026edae298">
regarding rate limit, for the above query only 10 records are there, no much rate limit is needed

**Comment by look** [comment, marked as answer]
@kaushalgupta94 unfortunately at this time, the REST API does not support regular expressions like the web UI. It's on our roadmap to fix, but has been delayed due to higher priority items.

There's a dedicated discussion for it: https://github.com/orgs/community/discussions/9991
